### ISSUE TITLE
Validation error popup has wrong size

### ISSUE THREAD

**Issue by WolfgangHG** [eclipse-lsp4e/lsp4e#214]
This is a followup to eclipse/wildwebdeveloper#848 - I was told to post it at lsp4e.

The following problem can be reproduced with the setting "Enable schema based validation = Always" and this xml snippet:

```
<?xml version="1.0" encoding="UTF-8"?>
<jboss-web xmlns="http://www.jboss.com/xml/ns/javaee"
           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
           xsi:schemaLocation="http://www.jboss.com/xml/ns/javaee http://some.invalid.host.com"
           version="12.1">
</jboss-web>
```

The popup for the validation error is too small (but my screen is wide enough at 1920x1280) - the error message is not shown completely, and the popup does not have a horizontal scrollbar:
![validationerror](https://user-images.githubusercontent.com/138049/184534091-8a244c88-3d12-4147-8360-c04174a037b8.png)

It took me some time until I noticed that the popup can be resized. When making it large enough so that the full text is shown, a vertical scrollbar appears.
![validationerror_resized](https://user-images.githubusercontent.com/138049/184534162-61ef4bf1-b20f-492e-8d06-829f5db025e3.png)

Seems there is a resizing issue.

**Comment by mickaelistria** [comment]
Can you please try with latest release of LSP4E installed?

**Comment by WolfgangHG** [comment]
Same with latest WildWebDeveloper and updating to the latest "lsp4e" snapshot "Language Server Protocol client for Eclipse IDE (Incubation)" version 0.16.2.202305251416.

**Comment by sebthom** [comment]
I think the root cause is an issue with [org.eclipse.ui.internal.genericeditor.hover.CompositeTextHover](https://github.com/eclipse-platform/eclipse.platform.ui/blob/master/bundles/org.eclipse.ui.genericeditor/src/org/eclipse/ui/internal/genericeditor/hover/CompositeTextHover.java).
The moment two different markers are available for a code location the rendered hover gets distorted. 

In your example the first line of text comes from a `org.eclipse.lsp4e.operations.diagnostics.DiagnosticAnnotation` translated to a `org.eclipse.ui.internal.genericeditor.markers.MarkerAnnotationHover` and the text below comes from a `org.eclipse.lsp4e.operations.hover.LSPTextHover`. Both generated by the Lemminx LS.

So I believe this needs to be fixed upstream in https://github.com/eclipse-platform/eclipse.platform.ui

**Comment by WolfgangHG** [comment]
Does this mean I should create a new issue in the other repo?

**Comment by sebthom** [comment]
I am not really familiar with the platform ui code base but I guess so.

**Comment by WolfgangHG** [comment]
Well, I cannot reproduce the initial issue with current Eclipse 2024-12, so I close this one.
Currently, I observer only one error/message with the sample.  Seems the "schemaLocation" info from my initial screenshot is not rendered currently.
![grafik](https://github.com/user-attachments/assets/b04ececb-efce-4e04-99d5-bd6257e520ff)

**Comment by rubenporras** [comment]
@sebthom, Yes, it is a bug in the platform. As far as I remember when IInformationControl is created inside the
CompositeInformationControl.createContent, the parent of its children are reassigned,
effectively making getShell() return an empty point
during computeSizeHint().
